yt_dlp/extractor/prankcast.py

```diff
@@ -0,0 +1,79 @@
+import json
+from datetime import datetime
+from .common import InfoExtractor
+from ..utils import parse_duration
+
+
+class PrankCastIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?prankcast\.com/.*/showreel/(?P<id>[0-9]+)'
+    _TESTS = [{
+        'url': 'https://prankcast.com/Devonanustart/showreel/1561-Beverly-is-back-like-a-heart-attack-',
+        'info_dict': {
+            'id': '1561',
+            'ext': 'mp3',
+            'title': 'Beverly is back like a heart attack!',
+            'uploader': 'Devonanustart',
+            'upload_date': '20220825'
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        # Extract the JSON
+        json_info = self._search_nextjs_data(webpage, video_id)
+
+        # Get the broadcast URL and the recording hash.
+        # The full URL is {broadcast_url}/{recording_hash}.mp3
+        broadcast_url = json_info['props']['pageProps']['ssr_data_showreel']['broadcast_url']
+        recording_hash = json_info['props']['pageProps']['ssr_data_showreel']['recording_hash']
+        url = broadcast_url + recording_hash + ".mp3"
+
+        # Get dates
+        start_date = None
+        if 'start_date' in json_info['props']['pageProps']['ssr_data_showreel']:
+            start_date = json_info['props']['pageProps']['ssr_data_showreel']['start_date'].replace('Z', '')
+
+        end_date = None
+        if 'end_date' in json_info['props']['pageProps']['ssr_data_showreel']:
+            end_date = json_info['props']['pageProps']['ssr_data_showreel']['end_date'].replace('Z', '')
+
+        # Get broadcast date
+        upload_date = None
+        if start_date is not None:
+            upload_date = start_date.split('T')[0].replace('-', '')
+
+        # Get broadcast title
+        broadcast_title = json_info['props']['pageProps']['ssr_data_showreel']['broadcast_title']
+
+        # Get author (AKA show host)
+        uploader = None
+        if 'user_name' in json_info['props']['pageProps']['ssr_data_showreel']:
+            uploader = json_info['props']['pageProps']['ssr_data_showreel']['user_name']
+
+        # Get the co-hosts/guests
+        if uploader is not None:
+            guests = [uploader]
+        else:
+            guests = []
+
+        if 'guests_json' in json_info['props']['pageProps']['ssr_data_showreel']:
+            for guest in json.loads(json_info['props']['pageProps']['ssr_data_showreel']['guests_json']):
```

Comment:
I think `_parse_json` can be used here instead of `json.loads`. But i think `json_info['props']['pageProps']['ssr_data_showreel']['guests_json']` already in json, no ?

Reply:
Changed it to _parse_json but I do have to parse it again. Otherwise I get a string and can't use the loop on it.

On the original json it looks like this:
`"guests_json":"[{\"id\": 2, \"date\": \"2022-08-08T23:56:19.000Z\", \"name\": \"dragonmere\", \"open\": 0, \"type\": \"cohost\", \"timestamp\": \"\", \"permissions\": \"0\"}, {\"id\": 14, \"date\": \"2022-08-08T23:56:25.000Z\", \"name\": \"splink\", \"open\": 0, \"type\": \"guest\", \"timestamp\": \"\", \"permissions\": \"0\"}, {\"id\": 19, \"date\": \"2022-08-08T23:56:35.000Z\", \"name\": \"wastedmemory\", \"open\": 0, \"type\": \"guest\", \"timestamp\": \"\", \"permissions\": \"0\"}, {\"id\": 585, \"date\": \"2022-08-09T00:57:36.000Z\", \"name\": \"bread336\", \"open\": 0, \"type\": \"guest\", \"timestamp\": \"\", \"permissions\": \"0\"}, {\"id\": 373, \"date\": \"2022-08-09T01:27:34.000Z\", \"name\": \"criefaux\", \"open\": 0, \"type\": \"guest\", \"timestamp\": \"\", \"permissions\": \"0\"}, {\"id\": 277, \"date\": \"2022-08-09T01:27:42.000Z\", \"name\": \"kbbles\", \"open\": 0, \"type\": \"guest\", \"timestamp\": \"\", \"permissions\": \"0\"}]",`

Reply:
Could you try `_parse_json` with argument `transform_source=js_to_json`? Be sure to import `js_to_json` first

Reply:
Sorry, I got it wrong on the last comment. Using _parse_json works great (without needing the js_to_json), not sure why it was giving me a string before...